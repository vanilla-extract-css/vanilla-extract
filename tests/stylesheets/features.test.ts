import {
  getStylesheet,
  startFixture,
  TestServer,
} from '@vanilla-extract-private/test-helpers';

const workerIndex = parseInt(process.env.JEST_WORKER_ID ?? '', 10);

const buildTypes = ['vite', 'esbuild', 'mini-css-extract', 'parcel'] as const;

buildTypes.forEach((buildType, index) => {
  describe(`features - ${buildType}`, () => {
    let server: TestServer;

    beforeAll(async () => {
      const portRange = 100 * workerIndex;

      server = await startFixture('features', {
        type: buildType,
        mode: 'production',
        basePort: 12000 + portRange + index,
      });
    });

    test('should create valid stylesheet', async () => {
      expect(
        await getStylesheet(server.url, server.stylesheet),
      ).toMatchSnapshot();
    });

    afterAll(async () => {
      await server.close();
    });
  });
});
