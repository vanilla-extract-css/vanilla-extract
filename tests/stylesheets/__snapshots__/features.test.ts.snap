// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features - esbuild should create valid stylesheet 1`] = `
"._1o6ek500 {
  height: 50px;
}
._1o6ek500:after {
  content: \\"Below 700px\\";
  display: block;
}
._1o6ek501 {
  background-color: #b0e0e6;
}
._1o6ek501:hover {
  background-color: #708090;
}
._1o6ek502 {
  background-color: #b0e0e6;
}
._1o6ek502:hover {
  background-color: #708090;
}
._1o6ek503 {
  background-color: #b0e0e6;
}
._1o6ek503:hover {
  background-color: #708090;
}
._1o6ek505 {
  color: #fff;
}
._1o6ek506 {
  background-color: #000;
}
body ._1o6ek507 {
  font-size: 24px;
}
._1o6ek508 {
  color: #fff;
}
._1o6ek509 {
  background-color: #000;
}
body ._1o6ek50a {
  font-size: 24px;
}
@media screen and (min-width: 700px) {
  ._1o6ek500 {
    color: plum;
  }
  ._1o6ek500:after {
    content: \\"Above 700px\\";
  }
}
"
`;

exports[`features - mini-css-extract should create valid stylesheet 1`] = `
"._87f2ru0 {
  height: 50px;
}
._87f2ru0:after {
  content: \\"Below 700px\\";
  display: block;
}
._87f2ru1 {
  background-color: #b0e0e6;
}
._87f2ru1:hover {
  background-color: #708090;
}
._87f2ru2 {
  background-color: #b0e0e6;
}
._87f2ru2:hover {
  background-color: #708090;
}
._87f2ru3 {
  background-color: #b0e0e6;
}
._87f2ru3:hover {
  background-color: #708090;
}
._87f2ru5 {
  color: #fff;
}
._87f2ru6 {
  background-color: #000;
}
body ._87f2ru7 {
  font-size: 24px;
}
._87f2ru8 {
  color: #fff;
}
._87f2ru9 {
  background-color: #000;
}
body ._87f2rua {
  font-size: 24px;
}
@media screen and (min-width: 700px) {
  ._87f2ru0 {
    color: plum;
  }
  ._87f2ru0:after {
    content: \\"Above 700px\\";
  }
}
"
`;

exports[`features - vite should create valid stylesheet 1`] = `
"._1o6ek500 {
  height: 50px;
}
._1o6ek500:after {
  content: \\"Below 700px\\";
  display: block;
}
._1o6ek501 {
  background-color: #b0e0e6;
}
._1o6ek501:hover {
  background-color: #708090;
}
._1o6ek502 {
  background-color: #b0e0e6;
}
._1o6ek502:hover {
  background-color: #708090;
}
._1o6ek503 {
  background-color: #b0e0e6;
}
._1o6ek503:hover {
  background-color: #708090;
}
._1o6ek505 {
  color: #fff;
}
._1o6ek506 {
  background-color: #000;
}
body ._1o6ek507 {
  font-size: 24px;
}
._1o6ek508 {
  color: #fff;
}
._1o6ek509 {
  background-color: #000;
}
body ._1o6ek50a {
  font-size: 24px;
}
@media screen and (min-width: 700px) {
  ._1o6ek500 {
    color: plum;
  }
  ._1o6ek500:after {
    content: \\"Above 700px\\";
  }
}
"
`;
